flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/JobSubmitTest.java

```diff
@@ -192,4 +193,78 @@ public void initializeOnMaster(ClassLoader loader) throws Exception {
 			fail(e.getMessage());
 		}
 	}
+
+	/**
+	 * Verifies failure when the client version is lower than supported.
+	 */
+	@Test
```

Comment:
You can implement the test with `@Test(expected=JobSubmissionException.class)` which checks that the correct exception is produced.